File: src/main/java/ru/citeck/ecos/ui/ListCellRendererWithIcon.java
package ru.citeck.ecos.ui;

import javax.swing.*;

public class ListCellRendererWithIcon extends DefaultListCellRenderer {

    private final Icon icon;
    private final int iconTextGap;

    public ListCellRendererWithIcon(Icon icon) {
        this.icon = icon;
        this.iconTextGap = 4;
    }

    @Override
    public Icon getIcon() {
        return icon;
    }

    @Override
    public int getIconTextGap() {
        return iconTextGap;
    }

}

File: src/main/kotlin/ru/citeck/idea/view/list/ListCellRendererWithIcon.kt
package ru.citeck.idea.view.list

import javax.swing.DefaultListCellRenderer
import javax.swing.Icon

class ListCellRendererWithIcon(private val icon: Icon) : DefaultListCellRenderer() {

    private val iconTextGap = 4

    override fun getIcon(): Icon {
        return icon
    }

    override fun getIconTextGap(): Int {
        return iconTextGap
    }
}
